xcodeproj/internal/xcodeprojinfo.bzl

```diff
@@ -256,6 +277,19 @@ def _skip_target(*, target, deps, deps_attrs, transitive_infos):
         ),
     )
 
+def _create_test_envs_depset(*, automatic_target_info, ctx, id, target):
+    test_env = getattr(ctx.rule.attr, automatic_target_info.env, {})
+    raw_run_env = target[RunEnvironmentInfo].environment if RunEnvironmentInfo in target else {}
+
+    # Some keys are not applicable in schemes, we will filter them out here
+    run_env = {}
+    denylist_run_env_keys = ["XCODE_VERSION_OVERRIDE", "XCODE_VERSION"]
```

Comment:
@brentleyjones Not sure what is the right move here. Without this filtering we cannot test across multiple xcode versions at once since the will mutate the fixtures for each xcode version.